homeassistant/components/epic_games_store/config_flow.py

```diff
@@ -0,0 +1,102 @@
+"""Config flow for Epic Games Store integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from epicstore_api import EpicGamesStoreAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import CONF_LOCALE, CONF_SUPPORTED_LOCALES, DOMAIN
+from .coordinator import not_handle_service_errors
+from .helper import get_country_from_locale
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_LOCALE): vol.In(CONF_SUPPORTED_LOCALES),
+    }
+)
+
+
+def get_default_locale(hass: HomeAssistant) -> str:
+    """Get default locale code based on Home Assistant config."""
+    locale_code = f"{hass.config.language}-{hass.config.country}"
+    if locale_code in CONF_SUPPORTED_LOCALES:
+        return locale_code
+    if hass.config.language in CONF_SUPPORTED_LOCALES:
+        return hass.config.language
+    return "en-US"
+
+
+async def validate_input(
```

Comment:
I'm a little bit in doubt with this function, I got the feeling this can be a lot shorter as we only check if a call can succeed

Reply:
Does "we only check if a call can succeed" and "I got the feeling this can be a lot shorter" be in the same sentence ? 🤣

There are actually 2 checks:
1. The API call
2. Test that the results has interested items

99% of integrations are only getting the first check ... (at least 100% of integrations I did 😝, and was merged).



Maybe you thought of instancing a coordinator here an make a first fetch and getting data ?
I thought of this too, seen few integrations did it, but don't know if it's really necessary.
And the coordinator constructor should be changed to not use a config entry as a parameter, and we have to close the coordinator so it not scheduled data fetch itself right ?


Also, logs and exceptions here may be more adapted to the config flow.


I think we can leave it that way for now and follow with something better afterwards.

Reply:
Proposal: What if we change the signature of this method to return a boolean? We are already putting in the LOCALE, so there's no real need to return that.

Alternative would be to return nothing and raise if it isn't correct. Not sure which approach I like more.